src/librustdoc/passes/collect_trait_impls.rs

```diff
@@ -90,7 +91,35 @@ crate fn collect_trait_impls(krate: Crate, cx: &DocContext<'_>) -> Crate {
         for &impl_node in cx.tcx.hir().trait_impls(trait_did) {
             let impl_did = cx.tcx.hir().local_def_id(impl_node);
             cx.tcx.sess.time("build_local_trait_impl", || {
-                inline::build_impl(cx, None, impl_did.to_def_id(), None, &mut new_items);
+                let mut extra_attrs = Vec::new();
+                let mut parent = cx.tcx.parent(impl_did.to_def_id());
+                while let Some(did) = parent {
+                    extra_attrs.extend(
+                        cx.tcx
+                            .get_attrs(did)
+                            .iter()
+                            .filter(|attr| attr.has_name(sym::doc))
+                            .filter(|attr| {
+                                (try {
```

Comment:
What's the point of this `try`?

Reply:
To allow using `?` on the `meta_item_list` return... though now that I changed how the checking for a single element works I might be able to do that as part of the `if let` instead...

Reply:
Yes, it seems a bit overkill in here. ;)